atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -448,10 +448,11 @@ private static LockAndTimestampServices withRefreshingLockService(
 
     private static LockAndTimestampServices withRequestBatchingTimestampService(
             java.util.function.Supplier<TimestampClientConfig> timestampClientConfigSupplier,
-            LockAndTimestampServices lockAndTimestampServices) {
-        TimelockService timelockServiceWithBatching =
+            LockAndTimestampServices lockAndTimestampServices, String userAgent) {
+        TimelockService timelockServiceWithBatching = ServiceCreator.createInstrumentedService(
                 DecoratedTimelockServices.createTimelockServiceWithTimestampBatching(
-                        lockAndTimestampServices.timelock(), timestampClientConfigSupplier);
+                        lockAndTimestampServices.timelock(), timestampClientConfigSupplier),
+                TimelockService.class, userAgent);
```

Comment:
@nziebart I think that's actually another code path. Would be good to have some unit tests for this maze of builders & wrappers, and we could assert that Instrumentation is correct. If we did double wrap, we'd be double counting execution counts and throw off other metrics.

Note to self: don't try reviewing on phone by the pool on vacation.